### PR TITLE
DROOLS-3815 Remove code duplicity in kie-dmn-validation

### PR DESCRIPTION
**Pull request by baldimir**
@tarilabs could you please review?

### CODE REVIEW

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java [hunk 1]
```diff
@@ -289,18 +251,43 @@ private void compileTableOutputClauses(DMNModel model, DecisionTable dt, DDTATab
         }
     }
 
+    private void checkMultipleConstraintsOnColumn(UnaryTestListNode utln, DecisionTable dt) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java]: Please rename `verifyUnaryTestsAllEQ`

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java [hunk 2]
```diff
@@ -289,18 +251,43 @@ private void compileTableOutputClauses(DMNModel model, DecisionTable dt, DDTATab
...
+        if (!utln.getElements().stream().allMatch(e -> e instanceof UnaryTestNode && ((UnaryTestNode) e).getOperator() == UnaryOperator.EQ)) {
+            throw new DMNDTAnalysisException("Multiple constraint on column: " + utln, dt);
+        }
+    }
+
+    private List<Comparable<?>> getDiscreteValues(UnaryTestListNode utln) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java]: Please rename `getDiscreteValuesSorted`

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java [hunk 3]
```diff
@@ -423,6 +387,26 @@ private static void findGaps(DTAnalysis analysis, DDTATable ddtaTable, int jColI
         LOG.debug(".");
     }
 
+    private static List<Bound> findBounds(DDTATable ddtaTable, int jColIdx, Collection<Integer> activeRules) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java]: Please rename `findBoundsSorted`

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java [hunk 4]
```diff
@@ -423,6 +387,26 @@ private static void findGaps(DTAnalysis analysis, DDTATable ddtaTable, int jColI
...
+            intervals = intervals.stream().filter(i -> activeRules.contains(i.getRule())).collect(Collectors.toList());
+        }
+        LOG.debug("intervals {}", intervals);
+        List<Bound> bounds = intervals.stream().flatMap(i -> Stream.of(i.getLowerBound(), i.getUpperBound())).collect(Collectors.toList());
+        Collections.sort(bounds);
+        LOG.debug("bounds (sorted) {}", bounds);
+        return bounds;
+    }
+
+    private static List<Interval> getEdgesFromIntervals(Interval[] currentIntervals, int intervalsIndex) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java]: Please rename `buildEdgesForHyperrectangleFromIntervals`

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java [hunk 5]
```diff
@@ -527,6 +495,16 @@ private static Interval lastDimensionUncoveredInterval(Bound<?> l, Bound<?> r, I
         return results;
     }
 
+    private static Interval createInterval(List discreteValues, int rule, int col, int lowerBoundIdx, int upperBoundIdx) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/DMNDTAnalyser.java]: Please rename `createIntervalOfRule`

**Review by tarilabs — APPROVED** [review]
If you can kindly accept the rename changes, because they would help me during debug.
Otherwise looks good and thanks a lot Tibor!

**Comment by baldimir** [comment]
@tarilabs methods renamed according to your feedback.

**Pull request merged**